### PR TITLE
patchValidation should consider kind if singular name is empty

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:

This fix a case when one don't want to provide singular name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-operator/resource/generate/components/crds.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ func addFieldsToAllVersions(crd *extv1.CustomResourceDefinition, fields ...inter
 
 func patchValidation(crd *extv1.CustomResourceDefinition, version *extv1.CustomResourceDefinitionVersion) error {
 	name := crd.Spec.Names.Singular
+	if name == "" {
```
Comment (rmohr) [inline pkg/virt-operator/resource/generate/components/crds.go:94]: @xpivarc could you add a test?

What was the case where you encountered this?

Comment (xpivarc) [reply pkg/virt-operator/resource/generate/components/crds.go:94]: This is the case https://github.com/kubevirt/kubevirt/pull/6026#discussion_r668521320 . I can have a look on what we can test

Comment (davidvossel) [reply pkg/virt-operator/resource/generate/components/crds.go:94]: so, this issue doesn't actually exist today then, right?  If not, maybe we should just fail the validation if no singular name is provided for the crd to ensure this doesn't occur. That would force the crd definitions to always have singular set or CI will fail.

Comment (xpivarc) [reply pkg/virt-operator/resource/generate/components/crds.go:94]: There isn't a issue. But you can have CRD without the `Singular`. In that case the lowercase `kind` is used... So I would say we can allow it by this PR.

**Comment by xpivarc** [comment]
@rmohr @davidvossel PTAL

**Comment by davidvossel** [comment]
>  @davidvossel PTAL

i still don't get what we're solving. why are we concerned about someone creating a crd without a singular name?

**Comment by xpivarc** [comment]
> > @davidvossel PTAL
> 
> i still don't get what we're solving. why are we concerned about someone creating a crd without a singular name?

According to docs, it is a valid case. I can make it fail if we don't want crds without a singular names.

**Comment by rmohr** [comment]
@xpivarc do you still think we want this in? Since we only want to patch our objects and there we provide the singular too, I am not sure if we need it.

**Comment by xpivarc** [comment]
> @xpivarc do you still think we want this in? Since we only want to patch our objects and there we provide the singular too, I am not sure if we need it.

Correct. This does only cover a case when we are introducing new objects to the project. One can omit the singular as it is optional and find himself finding why something is failing. This would cover the case, nothing more.

I think we can cover it by reviews but I will leave it up to your decision, feel free to close.

**Pull request closed without merging**